public/app/plugins/panel/datagrid/state.ts

```diff
@@ -30,56 +30,110 @@ interface DatagridState {
+  };
 }
 
-interface OpenCellContextMenuPayload {
-  event: CellClickedEventArgs;
-  cell: Item;
+interface OpenCellContextMenuAction {
+  type: DatagridActionType.openCellContextMenu;
+  payload: {
+    event: CellClickedEventArgs;
+    cell: Item;
+  };
 }
 
-interface OpenHeaderContextMenuPayload {
-  event: HeaderClickedEventArgs;
-  columnIndex: number;
+interface OpenHeaderContextMenuAction {
+  type: DatagridActionType.openHeaderContextMenu;
+  payload: {
+    event: HeaderClickedEventArgs;
+    columnIndex: number;
+  };
 }
 
-interface OpenHeaderDropdownMenuPayload {
-  screenPosition: Rectangle;
-  columnIndex: number;
-  value: string;
+interface OpenHeaderDropdownMenuAction {
+  type: DatagridActionType.openHeaderDropdownMenu;
+  payload: {
+    screenPosition: Rectangle;
+    columnIndex: number;
+    value: string;
+  };
 }
 
-interface ColumnMovePayload {
-  from: number;
-  to: number;
+interface ColumnMoveAction {
+  type: DatagridActionType.columnMove;
+  payload: {
+    from: number;
+    to: number;
+  };
 }
 
-interface MultipleCellsSelectedPayload {
-  selection: GridSelection;
+interface MultipleCellsSelectedAction {
+  type: DatagridActionType.multipleCellsSelected;
+  payload: {
+    selection: GridSelection;
+  };
 }
 
-interface ColumnFreezeChangedPayload {
-  columnIndex: number;
+interface ColumnFreezeChangedAction {
+  type: DatagridActionType.columnFreezeChanged;
+  payload: {
+    columnIndex: number;
+  };
 }
 
-export interface DatagridAction {
-  type: DatagridActionType;
-  payload?:
-    | UpdateColumnsPayload
-    | ColumnResizeStartPayload
-    | OpenCellContextMenuPayload
-    | OpenHeaderContextMenuPayload
-    | OpenHeaderDropdownMenuPayload
-    | ColumnMovePayload
-    | MultipleCellsSelectedPayload
-    | ColumnFreezeChangedPayload;
+interface ColumnResizeEndAction {
+  type: DatagridActionType.columnResizeEnd;
 }
+interface ShowColumnRenameInputAction {
+  type: DatagridActionType.showColumnRenameInput;
+}
+interface HideColumnRenameInputAction {
+  type: DatagridActionType.hideColumnRenameInput;
+}
+interface CloseContextMenuAction {
+  type: DatagridActionType.closeContextMenu;
+}
+interface CloseSearchAction {
+  type: DatagridActionType.closeSearch;
+}
+interface OpenSearchAction {
+  type: DatagridActionType.openSearch;
+}
+interface GridSelectionClearedAction {
+  type: DatagridActionType.gridSelectionCleared;
+}
+interface ColumnFreezeResetAction {
+  type: DatagridActionType.columnFreezeReset;
+}
+
+export type DatagridAction =
+  | UpdateColumnsAction
+  | ColumnResizeStartAction
+  | OpenCellContextMenuAction
+  | OpenHeaderContextMenuAction
+  | OpenHeaderDropdownMenuAction
+  | ColumnMoveAction
+  | MultipleCellsSelectedAction
+  | ColumnFreezeChangedAction
+  | ColumnResizeEndAction
+  | ShowColumnRenameInputAction
+  | HideColumnRenameInputAction
+  | CloseContextMenuAction
+  | CloseSearchAction
+  | OpenSearchAction
+  | GridSelectionClearedAction
+  | ColumnFreezeResetAction;
```

Comment:
All of these types are unnecessary if we used redux toolkit

Reply:
except the payload types :)

Reply:
Completely true - this probably would be a whole lot easier with redux toolkit, but I'll leave that up to the code owners to do. I just wanted to come in and fix these with some quick and trival changes :)

Reply:
Agreed, it's on my todolist to refactor this in a better way